src/de/jost_net/JVerein/gui/dialogs/BuchungenMitgliedskontenZuordnungDialog.java

```diff
@@ -0,0 +1,535 @@
+    useIban = new CheckboxInput(settings.getBoolean(SETTINGS_NAME_IBAN, true));
+    useMemberNumber = new CheckboxInput(settings.getBoolean(SETTINGS_NAME_MITGLIEDSNUMMER, false));
+    useName = new CheckboxInput(settings.getBoolean(SETTINGS_NAME_VORNAME_NAME, false));
+  }
+
+  private DateInput createDateInput(Date date, boolean isStart)
+  {
+    DateInput returnValue = new DateInput(date, new JVDateFormatTTMMJJJJ());
+    String typeOfInput = isStart ? "Be�ginn" : "Ende";
+    returnValue.setTitle(typeOfInput +  " des Suchbereichs");
+    returnValue.setText("Bitte "+ typeOfInput +" des Suchbereichs w�hlen");
+    returnValue.setComment("*)");
+    return returnValue;
+  }
+
+  /**
+   * @see de.willuhn.jameica.gui.dialogs.AbstractDialog#paint(org.eclipse.swt.widgets.Composite)
+   */
+  @Override
+  protected void paint(Composite parent) throws Exception
+  {
+    Container group = new SimpleContainer(parent);
+    group.addText("Bitte w�hlen Sie den Suchzeitraum und die gew�nschte Zuordnungsart aus", true);
+
+    group.addLabelPair("Startdatum", dateFrom);
+    group.addLabelPair("Enddatum", dateUntil);
+    group.addLabelPair("nach eindeutiger IBAN", useIban);
+    group.addLabelPair("nach " + (Einstellungen.getEinstellung().getExterneMitgliedsnummer().booleanValue() ? "Ext. Mitgliedsnummer" : "Mitgliedsnummer"), useMemberNumber);
+    group.addLabelPair("nach eindeutigen Vorname und Nachname", useName);
+    ButtonArea buttons = new ButtonArea();
+
+    Button button = new Button("Zuordnungen suchen", new Action()
+    {
+
+      @Override
+      public void handleAction(Object context) throws ApplicationException
+      {
+        doSearchAssignment();
+      }
+    }, null, true, "user-friends.png");
+    buttons.addButton(button);
+    buttons.addButton("Abbrechen", new Action()
+    {
+
+      @Override
+      public void handleAction(Object context)
+      {
+        throw new OperationCanceledException();
+      }
+    }, null, false, "stop-circle.png");
+    group.addButtonArea(buttons);
+    getShell()
+        .setMinimumSize(getShell().computeSize(WINDOW_WIDTH, SWT.DEFAULT));
+  }
+
+  /**
+   * Ermittelt eine Liste von Buchungen mit Mitgliedskonten und gibt
+   * diese an den BuchungenMitgliedskontenZuordnungVorschauDialog weiter
+   * 
+   */
+  private void doSearchAssignment()
+  {
+    final Date dateFromInput = (Date)dateFrom.getValue();
+    final Date dateUntilInput = (Date)dateUntil.getValue();
+
+    final boolean useIbanInput = Boolean.TRUE.equals(this.useIban.getValue());
+    final boolean useMemberNumberInput = Boolean.TRUE.equals(this.useMemberNumber.getValue());
+    final boolean useNameInput = Boolean.TRUE.equals(this.useName.getValue());
+
+    settings.setAttribute(SETTINGS_NAME_IBAN, useIbanInput);
+    settings.setAttribute(SETTINGS_NAME_MITGLIEDSNUMMER, useMemberNumberInput);
+    settings.setAttribute(SETTINGS_NAME_VORNAME_NAME, useNameInput);
+
+    BackgroundTask t = new BackgroundTask()
```

Comment:
Personally I would move this application logic out of the UI control implementation into a separate class. It would be
1. a clearer separation of concerns (UI vs. Abgleichlogik)
2. writing unit tests for the Abgleichslogik would be easier.

But that is only a suggestion. It's better to have this logic here than not having it at all :-)